//python3 -m grpc_tools.protoc -I protos --python_out=. --grpc_python_out=. protos/fet.proto
syntax = "proto3";

package com.groupseven.fet;

service MasterClientCommunicationService {

    rpc GetRandomSplitValueFromClient(GetRandomSplitValueFromClientRequest) returns (GetRandomSplitValueFromClientResponse);

    rpc GetAggregatedValuesFromClient(GetAggregatedValuesFromClientRequest) returns (GetAggregatedValuesFromClientResponse);

    rpc BroadcastTreeNodesBasedOnBestSplit(BroadcastTreeNodesBasedOnBestSplitRequest) returns (ReceivedResponse);

}

message ReceivedResponse {
    bool response = 1;
}

message GetRandomSplitValueFromClientRequest {
    string feature = 1;
    int32 clientId = 2;
}

message GetRandomSplitValueFromClientResponse {
    int32 clientId = 1;
    double splitValue = 2;
}

message GetAggregatedValuesFromClientRequest {
    string feature = 1;
    int32 clientId = 2;
    double splitValue = 3;
    string dataSet = 4;
}

message GetAggregatedValuesFromClientResponse {
    AggregatedValue aggregatedValueLeft = 1;
    AggregatedValue aggregatedValueRight = 2;
}

message AggregatedValue {
    int32 approvedLoan = 1;
    int32 declinedLoan = 2;
}

message BroadcastTreeNodesBasedOnBestSplitRequest {
    string feature = 1;
    double splitValue = 2;
    int32 treeHeight = 3;
    string dataSet = 4;
}